### PR TITLE
Use Pinia State for Modals

### LINKED ISSUE #212: SearchResultDetails - Fix close button functionality
# Context

Fix button isn't always closing the window. Needs to work across browsers and OSes. 

# Requirements

- [x] fix it
- [x] ensure it works on windows and mac. 
- [x] Ensure it works in safari and chrome (at least)

# Dependencies

- None (Add any issue dependencies here)

### PR DESCRIPTION
**Pull request by theyokohamalife**
Resolves #212 

# What changed
1. @pinia/nuxt was added to better facilitate using Pinia to manage the state for components.
2. modalStore.ts was created to manage the Modal.vue component's state
3.  `.env` was added to .gitignore.

# Testing instructions
1. Rename app.vue to app-backup.vue and  app-copy.vue to app.vue in order to see the main page.
4. Change `isOpen: false` to `isOpen: true` in _modalStore.ts_ so the modal opens by default when you refresh the page.
Screenshot 2023-07-17 at 7 17 04 PM

5. Start the dev server and confirm that clicking the X on the modal closes it on all Chrome, Firefox, and Safari.

### CODE REVIEW

File: components/MainContentContainer.vue [hunk 1]
```diff
@@ -4,9 +4,13 @@ import SearchResultDetails from './SearchResultDetails.vue';
 <template>
     <div class="flex flex-1 bg-secondary-bg/20 hover:shadow-inner hover:shadow-secondary-bg/90">
         <!-- <WelcomeSection /> -->
-        <Modal class="min-h-1/2 ml-8 mt-12">
+        <Modal v-show="useModalStore().$state.isOpen" class="min-h-1/2 ml-8 mt-12">
```
Comment (ermish) [inline components/MainContentContainer.vue:7]: you should initialize the store in the script section and then use it here.

File: components/Modal.vue [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <template>
     <div class="absolute z-10 bg-primary-bg rounded-xl overflow-hidden
         hover:shadow-inner hover:shadow-secondary-bg/90">
-        <div v-if="isOpen" class="modal">
-            <button @click="closeModal"
+        <div class="modal">
+            <button @click="useModalStore().hideModal()"
```
Comment (ermish) [inline components/Modal.vue:5]: you should initialize the store in the script section and then use it here.

File: .gitignore [hunk 3]
```diff
@@ -3,7 +3,10 @@
 .output
 dist
 
-#package management and yarn3
+# Environment files
+.env*
```
Comment (ermish) [inline .gitignore:7]: let's chat through this one. personally, I would recommend checking them in. Realistically your env file contains settings for each environment and then your secrets like passwords are added through runtime variables. (netlify variables for example)

**Review by ermish — CHANGES_REQUESTED** [review]
nice! easy changes

File: .gitignore (hunk 3 above)
Comment (theyokohamalife) [reply .gitignore:7]: Okay, I'll remove it for now, because I really want to get these changes merged.

**Review by ermish — APPROVED** [review]
LGTM

**Pull request merged**